[Jacob Ng] iP

Jacobnbh opened the pull request:
# DukeTheHelper
> "It is not good that man should be alone, I will make him a **_HELPER_** comparable to him."

DukeTheHelper is a helper bot that helps you to keep track of your tasks. It is:
* text-based
* easy to learn
* ~~FAST~~ SUPER FAST to use

All you need to do is,
1. download it from [here](https://github.com/Jacobnbh/ip)
2. double-click it.
3. add your tasks via commands
4. let it **HELP** you keep track of your tasks

And it is __*FREE*__

Features:
- [x] Managing 3 different types of tasks (todo, deadlines, events)
- [ ] Managing deadlines (coming soon)
- [ ] Reminders (coming soon)

If you would like some programming practice, you can also use it to practice your Java! Here's the `main` method:
````java
public class Main {
    public static void main(String[] args) {
        Application.launch(MainApp.class, args);
    }
}

Commit:
Add Gradle support

Jacobnbh committed:
Implement greetings and echo message function

Jacobnbh committed:
Add ability to store task and display list of tasks

Jacobnbh committed:
Add ability to mark tasks as done

Jacobnbh committed:
Add support for tracking ToDos, Deadlines and Events

Jacobnbh committed:
Update input.txt and expected.txt for automated text UI testing

Jacobnbh committed:
Update Duke.java to handle errors

Jacobnbh committed:
Update TaskList.java to support deleting of tasks from list and Duke.java to handle more errors

Jacobnbh committed:
Add ability to save and load tasks

Saves the task in the hard disk automatically whenever the task lit changes. Loads the data from the hard disk when Duke starts up.

Jacobnbh committed:
Fix error-handling bugs

Jacobnbh committed:
Add ability to understand dates

Jacobnbh committed:
Update ability to handle dates for Event type

Jacobnbh committed:
Merge branch 'branch-Level-7'

Jacobnbh committed:
Merge branch 'branch-Level-8'

# Conflicts:
#	src/main/java/Deadline.java
#	src/main/java/Duke.java
#	src/main/java/Event.java
#	src/main/java/TaskList.java

Jacobnbh committed:
Make the code more OOP

Jacobnbh committed:
Refactor into packages

Jacobnbh committed:
Add JUnit tests for Deadline and Event task types

Jacobnbh committed:
Add JavaDocs comments

Jacobnbh committed:
Update code to conform to coding standards

Jacobnbh committed:
Merge branch 'branch-A-JavaDoc'

Jacobnbh committed:
Merge branch 'branch-A-CodingStandard'

Jacobnbh committed:
Add ability for users to find a task by searching for a keyword

Jacobnbh committed:
Merge branch 'branch-Level-9'

Jacobnbh commented:
#DukeTheHelper
> It is not good that man should be alone, I will make him a helper comparable to him.

DukeTheHelper is a helper bot that helps you to keep track of your tasks. It is:
text-based
easy to learn
~~FAST~~ SUPER FAST to use

All you need to do is,
1. download it from [here] (https://github.com/Jacobnbh/ip)
2. double-click it.
3. add your tasks via commands
4. let it **HELP** you keep track of your tasks

And it is __*FREE*__

Features:
- [x] Managing 3 different types of tasks (todo, deadlines, events)
- [ ] Managing deadlines (coming soon)
- [ ] Reminders (coming soon)

If you would like some programming practice, you can also use it to practice your Java! Here's the `main` method:
````java
public class Main {
    public static void main(String[] args) {
        Application.launch(MainApp.class, args);
    }
}
````

jianoway commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,47 @@
+
+    public Duke(String filePath) {
+        this.ui = new Ui();
+        this.storage = new Storage("/data/duke.txt");
```
This isn't really a coding standard violation but could this instance of Storage be constructed with the filePath argument instead?

jianoway commented on src/main/java/duke/io/Storage.java:
```diff
@@ -0,0 +1,77 @@
+            }
+
+            switch (input[0]) {
+                case "T":
```
Perhaps you might consider un-indenting this (and it's children)?

jianoway commented on src/main/java/duke/io/Storage.java:
```diff
@@ -0,0 +1,77 @@
+
+    }
+
+    public void save(String textToAppend, boolean appendStatus) throws IOException {
```
Maybe this method name could be a little more descriptive? Same comment for refresh!

jianoway commented on src/main/java/duke/task/Task.java:
```diff
@@ -0,0 +1,26 @@
+ */
+public abstract class Task {
+    public abstract String showTask();
+    public abstract String showTaskOnly();
```
Maybe consider renaming this method to reflect that you're showing the task name only as the implementations seem to reflect this?

jianoway commented on src/main/java/duke/task/ToDo.java:
```diff
@@ -0,0 +1,35 @@
+
+    public String showWhen() { return ""; }
+
+    public String showTaskOnly() { return taskName; }
```
Possibly follow the Egyptian brackets style?

jianoway reviewed (commented):
Overall good job with the iP so far! The code is pretty clean but could do with some touchups such as being consistent with choosing to use periods or not in JavaDoc comments or some lines being quite long (close to 120 but not exceeding), making it slightly hard to read!

Jacobnbh committed:
Add GUI to Duke by implementing JavaFX technology

Jacobnbh committed:
Merge branch 'Level-10'

Jacobnbh committed:
Merge remote-tracking branch 'upstream/add-gradle-support'

Jacobnbh committed:
Add Gradle to automate build tasks of the project

Jacobnbh committed:
Merge branch 'branch-A-Gradle'

Siddharth-Sid commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,61 @@
+        try {
+            /* try loading text file from storage */
+            this.taskList = new TaskList(storage.load());
+        } catch (DukeException de) {
```
I am not sure it is convention or not but in general all exceptions caught as written as "e".

Siddharth-Sid commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,61 @@
+     *
+     * @return Duke's welcome message
+     */
+    public String welcomeMessage() {
```
Maybe you can remove this function and use the ui variable directly?

Siddharth-Sid commented on src/main/java/duke/task/TaskList.java:
```diff
@@ -0,0 +1,130 @@
+ */
+public class TaskList {
+    private ArrayList<Task> list;
+    private int counter = 0;
```
Since this counter is not directly accessible outside and is not be used for any check you could maybe remove it.

Siddharth-Sid reviewed (commented):
I think your code is well written with no errors. I have made a few suggestion for you to look at. None of them are necessary but you can take  a look.

Jacobnbh committed:
Add assertions to document important assumptions in the code

certain assumptions about the program state are not accounted for

updating the code with assert statements allows these assumptions
to be raised whenever it is met in the program. For example, if the
file path to the text file is empty, raise an Assertion Error.

having assertion statements raise errors indicates that there is a
possible bug and it is safer for the execution of the program stops
this way

Jacobnbh committed:
Merge pull request #1 from Jacobnbh/branch-A-Assertions

Add assertions to document important assumptions in the code

Jacobnbh committed:
Improve code quality

Improving code quality increases readability and understandability
for readers.

Refactoring, shortening, and simplifying the code by applying SLAP
(Single Layer of Abstraction) and other coding quality principles.

Jacobnbh committed:
Merge pull request #2 from Jacobnbh/branch-A-CodeQuality

Improve code quality

Jacobnbh committed:
Add ability to add and show snippets of notes for tasks

Jacobnbh committed:
Merge branch 'branch-D-Notes'

Jacobnbh committed:
Improve Duke GUI

Jacobnbh committed:
Set theme jekyll-theme-cayman

Jacobnbh committed:
Add a brief User Guide for Duke

Jacobnbh committed:
Add a brief User Guide

Jacobnbh committed:
Merge branch 'master' of https://github.com/Jacobnbh/ip

Jacobnbh committed:
testing

Jacobnbh committed:
Merge pull request #5 from Jacobnbh/branch-test

testing

Jacobnbh committed:
testing 2

Jacobnbh committed:
Merge pull request #6 from Jacobnbh/branch-test

testing 2

Jacobnbh committed:
testing 3

Jacobnbh committed:
Merge pull request #7 from Jacobnbh/branch-test

testing 3

Jacobnbh committed:
Revert "testing 3"

Jacobnbh committed:
Merge pull request #8 from Jacobnbh/revert-7-branch-test

Revert "testing 3"

Jacobnbh committed:
Revert "testing 2"

Jacobnbh committed:
Merge pull request #9 from Jacobnbh/revert-6-branch-test

Revert "testing 2"

Jacobnbh committed:
Revert "testing"

Jacobnbh committed:
Merge pull request #10 from Jacobnbh/revert-5-branch-test

Revert "testing"
